src/policy/compiler.rs

```diff
@@ -994,18 +995,25 @@ where
                 })
                 .collect();
 
-            if key_vec.len() == subs.len() && subs.len() <= MAX_PUBKEYS_PER_MULTISIG {
-                insert_wrap!(AstElemExt::terminal(Terminal::Multi(k, key_vec)));
-            }
-            // Not a threshold, it's always more optimal to translate it to and()s as we save the
-            // resulting threshold check (N EQUAL) in any case.
-            else if k == subs.len() {
-                let mut policy = subs.first().expect("No sub policy in thresh() ?").clone();
-                for sub in &subs[1..] {
-                    policy = Concrete::And(vec![sub.clone(), policy]);
+            match Ctx::sig_type() {
+                SigType::Schnorr => {
+                    insert_wrap!(AstElemExt::terminal(Terminal::MultiA(k, key_vec)))
+                }
+                _ => {
+                    if key_vec.len() == subs.len() && subs.len() <= MAX_PUBKEYS_PER_MULTISIG {
+                        insert_wrap!(AstElemExt::terminal(Terminal::Multi(k, key_vec)));
+                    }
+                    // Not a threshold, it's always more optimal to translate it to and()s as we save the
+                    // resulting threshold check (N EQUAL) in any case.
+                    else if k == subs.len() {
```

Comment:
We could also add this optimization to multi_a. If `k == n`, compiling as `ands() ` is more efficient.

Reply:
Updated in https://github.com/rust-bitcoin/rust-miniscript/pull/291/commits/1a1d19262b859c506d8a60baa3a5ac3643b0fe3f